### PR TITLE
ANN: separate color for function and method calls

### LINKED ISSUE #4039: Separate color scheme for functions
## Environment
* **Intellij-Rust plugin version:** v0.2.99.2126-191
* **Rust toolchain version:** 1.35.0
* **IDE name and version:** IntelliJ IDEA 2019.1
* **Operating system:** Windows 10 1903

## Problem description
rust

Currently, color scheme for function **declaration** and function **call** is same.
It would be great if the color highlighting option for function is separated to two options: declaration and call.

### PR DESCRIPTION
**Pull request by Undin**
Also:
* improve display names for some colors
* update highlighter demo

Closes #4039

### CODE REVIEW

File: src/main/kotlin/org/rust/ide/colors/RsColors.kt [hunk 1]
```diff
@@ -19,17 +19,18 @@ enum class RsColor(humanName: String, val default: TextAttributesKey? = null) {
...
+    FUNCTION_CALL("Functions//Function and method call", Default.FUNCTION_CALL),
     MACRO("Functions//Macro", Default.IDENTIFIER),
 
     PARAMETER("Parameters//Parameter", Default.PARAMETER),
     MUT_PARAMETER("Parameters//Mutable parameter", Default.PARAMETER),
     SELF_PARAMETER("Parameters//Self parameter", Default.KEYWORD),
     LIFETIME("Parameters//Lifetime", Default.IDENTIFIER),
     TYPE_PARAMETER("Parameters//Type parameter", Default.IDENTIFIER),
-    CONST_PARAMETER("Parameters//Const parameter", CONSTANT.textAttributesKey),
+    CONST_PARAMETER("Parameters//Const parameter", Default.CONSTANT),
```
Comment (vlad20012) [inline src/main/kotlin/org/rust/ide/colors/RsColors.kt]: Why is this changed?

Comment (Undin) [reply src/main/kotlin/org/rust/ide/colors/RsColors.kt]: As we discussed in person, constant parameters are more parameters then constants (for example, in C++ there is a separate color for template params including constant params) so it's quite unclear why the color for them should depend on constant color.
Also, it can be confusing that `CONST_PARAMETER` changes when you change `CONSTANT` but not vice versa

**Comment by Undin** [comment]
bors r=mchernyavsky
